02/app.js

```diff
@@ -1,16 +1,31 @@
-import React from 'react';
-import { createRoot } from 'react-dom/client';
+import React from "react";
+import { createRoot } from "react-dom/client";
 
-const root = createRoot(document.querySelector('#root'));
+const root = createRoot(document.querySelector("#root"));
 
 class Counter extends React.Component {
-    state = {
-        amount: 0,
-    }
-    
-    render() {
-        return <button>click me ({ this.state.amount })</button>
-    }
+  constructor(props) {
+    super(props);
+    this.state = { amount: 0 };
+  }
+
+  render() {
+    return <button ref={(button) => (this.button = button)}>click me ({this.state.amount})</button>;
+  }
+
+  componentDidMount() {
+    this.button.addEventListener("click", this.handleClick);
+  }
+
+  componentWillUnmount() {
+    this.button.removeEventListener("click", this.handleClick);
+  }
```

Comment:
Zdecydowanie lepiej (i prościej) jest napisać:  `<button onClick={this.handleClick}>`